security/pkg/server/ca/authenticate/kubeauth/kube_jwt.go

```diff
@@ -101,6 +102,9 @@ func (a *KubeJWTAuthenticator) authenticateHTTP(req *http.Request) (*security.Ca
 		return nil, fmt.Errorf("target JWT extraction error: %v", err)
 	}
 	clusterID := cluster.ID(req.Header.Get(clusterIDMeta))
+	if !features.RemoteClusterAccess {
+		clusterID = "" // do not allow other clusters unless Istiod is running in 'central istiod' mode.
```

Comment:
Should we silently check the same cluster? Or hard fail? The error may be strange error log vs a clear "cluster X requested, only local cluster allowed" log?

Reply:
I guess we can refine this - not sure, but I'm a bit afraid of breaking the 'good' case or confusing users who have a single cluster - but get cluster id meta because of injection or other things. 

The problem is that you can't know if it's really a remote cluster - or the local cluster with Kubernetes cluster ID, or the same cluster id that istio is configured with. Pretty messy. I tried few other variations.